Deploy/Functions/New-ADMUTemplate.ps1

```diff
@@ -231,6 +231,18 @@ if (`$PSBoundParameters.Count -eq 0) {
 "@
         # add executable region to the template
         $templateString += $executableRegion + [Environment]::NewLine
+        # replace the validation region [System.Management.Automation.ValidationMetadataException] lines with Write-ToLog Error Lines and exit 1
```

Comment:
Significant change here but when we were building the EXE from the template ps1 file, we found that the exe was never exiting with error code 1 when we were validating the migration parameters and re-requirement checks. 

We missed this because the invoke script generally took care of the same validation except for an invalid API key. 

So there's a change here when we build the template file to change the Throw lines with the `[System.Management.Automation.ValidationMetadataException]` to be Write-toLog Error messages and exit 1 for these conditions.